core/trino-main/src/main/java/io/trino/memory/MemoryPool.java

```diff
@@ -207,6 +207,19 @@ public boolean tryReserve(TaskId taskId, String allocationTag, long bytes)
         return true;
     }
 
+    public boolean tryReserve(long bytes)
```

Comment:
I do think we should also track memory usage for connectors as separate variable, `e.g: long connectorsReservedBytes` and expose it via `@Manager` getter method